src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -394,7 +394,8 @@ internal override ModuleSymbol ContainingModule
             get;
         }
 
-        internal override FileIdentifier? AssociatedFileIdentifier
+        internal sealed override bool IsFileLocal => AssociatedFileIdentifier is not null;
```

Comment:
Consider updating the `AssociatedFileIdentifier` property below to use `IsFileLocal`, to avoid duplication. `return IsFileLocal ? new FileIdentifier { ... } : null;`

Reply:
I considered some strategies for avoiding duplication in the below `AssociatedFileIdentifier` implementation, but in the end felt that the current solution was adequate.

Reply:
I didn't follow, what's the downside of the refactoring I proposed?

Reply:
Having to separately read the fields again in AssociatedFileIdentifier, and either add MemberNotNullAttribute to IsFileLocal or add ! when we know some are non-null.